Dockerfile

```diff
@@ -1,10 +1,24 @@
-FROM golang:1.12
+FROM golang:1.12-alpine as build
 
 WORKDIR /src
 
+# build-base is required for tests
+RUN apk add --no-cache \
+    build-base \
+    git
+
 COPY go.mod go.sum ./
 RUN go mod download
 
 COPY . .
 RUN GO111MODULE=on go install .
 CMD ["whalebrew"]
+
+FROM alpine
+COPY --from=docker:18.06 /usr/local/bin/docker /usr/local/bin/docker
+COPY --from=build /go/bin/whalebrew /usr/local/bin/whalebrew
+
+ENTRYPOINT ["whalebrew"]
```

Comment:
For optimum caching (especially when not using BuildKit (`DOCKER_BUILDKIT=1`)), you might want to move this `ENTRYPOINT` and the lines below above the `COPY` lines, so:

```Dockerfile
FROM alpine
ENTRYPOINT ["whalebrew"]
LABEL io.whalebrew.config.volumes '["/var/run/docker.sock:/var/run/docker.sock", "$WHALEBREW_INSTALL_PATH:$WHALEBREW_INSTALL_PATH"]'
LABEL io.whalebrew.config.keep_container_user 'true'
ENV PATH=${PATH}:/whalebrew

COPY --from=docker:18.06 /usr/local/bin/docker /usr/local/bin/docker
COPY --from=build /go/bin/whalebrew /usr/local/bin/whalebrew
```